### PR TITLE
fix(clients/go): update linter

### PR DESCRIPTION
**Pull request by Sijoma**
## Description

- golint is deprecated since v1.41.0
- it was replaced by revive

Error message:
`WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive. `

## Related issues

closes #

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by Sijoma** [comment]
Oh this adds some warnings, that I think I did not see locally. 🤔 

```
  Warning: var-naming: const OAuthClientIdEnvVar should be OAuthClientIDEnvVar (revive)
  Warning: var-naming: const OAuthAuthorizationUrlEnvVar should be OAuthAuthorizationURLEnvVar (revive)
  Warning: var-naming: method BPMNProcessId should be BPMNProcessID (revive)
  Warning: var-naming: method MessageId should be MessageID (revive)
  Warning: var-naming: method DecisionId should be DecisionID (revive)
```

**Comment by Sijoma** [comment]
Fixed some warnings like shown below and ran go fmt:

> pkg/zbc/oauthCredentialsCache_test.go:19:2: SA1019: "io/ioutil" has been deprecated since Go 1.16: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details. (staticcheck)
>         "io/ioutil"
>         ^
> pkg/zbc/oauthCredentialsProvider_test.go:29:2: SA1019: "io/ioutil" has been deprecated since Go 1.16: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details. (staticcheck)
>         "io/ioutil"
>         ^
> test/oauth_integration_test.go:25:2: SA1019: "io/ioutil" has been deprecated since Go 1.16: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details. (staticcheck)
>         "io/ioutil"
>         ^

File: clients/go/cmd/zbctl/internal/commands/cancel.go [hunk 1]
```diff
@@ -4,7 +4,7 @@
 // you may not use this file except in compliance with the License.
 // You may obtain a copy of the License at
 //
-//     http://www.apache.org/licenses/LICENSE-2.0
+//	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (Sijoma) [inline clients/go/cmd/zbctl/internal/commands/cancel.go:7]: No Idea why go fmt touched this or was it my editor 🤔 .

Comment (Zelldon) [reply clients/go/cmd/zbctl/internal/commands/cancel.go:7]: Are these tabs? Do you have maybe different setting for tabs 🤔 like converting to 4 spaces or something.

**Comment by npepinpe** [comment]
Linting is passing now :+1:

**Review by npepinpe — APPROVED** [review]
bors merge

**Pull request merged**